cupy/core/_routines_statistics.pyx

```diff
@@ -4,12 +4,24 @@ from cupy.core._kernel import ReductionKernel
 from cupy.core cimport _routines_math as _math
 from cupy.core.core cimport ndarray
 
+import cupy
+if cupy.cuda.cub_enabled:
+    from cupy.cuda import cub
+
 
 cdef ndarray _ndarray_max(ndarray self, axis, out, dtype, keepdims):
+    if cupy.cuda.use_cub():
+        if (cub.can_use_reduce_max(self.dtype, dtype) and (axis is None) and
```

Comment:
Please check `self.c_contiguous` and `self.f_contiguous` to correct access to array.

Reply:
I don't think that checking is necessary, because `core.ascontiguousarray` is called to make an array C-contiguous before calling actual cub routines.

Reply:
OK. I understand.